src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -87,129 +72,32 @@ private Assembly LoadFromPathUncheckedCore(string fullPath, AssemblyIdentity ide
                 loadedAssembly = LoadFromPathImpl(fullPath);
             }
 
-            // Add the loaded assembly to both path and identity cache.
-            return AddToCache(loadedAssembly, fullPath, identity);
-        }
-
-        private Assembly AddToCache(Assembly assembly, string fullPath, AssemblyIdentity identity)
-        {
-            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
-            Debug.Assert(assembly != null);
-
-            identity = AddToCache(fullPath, identity ?? AssemblyIdentity.FromAssemblyDefinition(assembly));
-            Debug.Assert(identity != null);
-
+            // Add the loaded assembly to the path cache.
             lock (_guard)
             {
-                // The same assembly may be loaded from two different full paths (e.g. when loaded from GAC, etc.),
-                // or another thread might have loaded the assembly after we checked above.
-                if (_loadedAssembliesByIdentity.TryGetValue(identity, out var existingAssembly))
-                {
-                    assembly = existingAssembly;
-                }
-                else
-                {
-                    _loadedAssembliesByIdentity.Add(identity, assembly);
-                }
-
-                // An assembly file might be replaced by another file with a different identity.
-                // Last one wins.
-                _loadedAssembliesByPath[fullPath] = assembly;
-
-                return assembly;
+                _loadedAssembliesByPath[fullPath] = loadedAssembly;
```

Comment:
This appears to be a bug if the goal is to ensure that two loads for the same path always return the same thing per the interface documentation, but it's been around since https://github.com/dotnet/roslyn/pull/11956. @mavasani or @tmat any chance if this is a load-bearing bug?